script/smoke_test.sh

```diff
@@ -3,9 +3,9 @@
 # This is an integration test to generally check the quality of parsing.
 
 SCALA_SCALA_LIBRARY_EXPECTED=95
-SCALA_SCALA_COMPILER_EXPECTED=84
+SCALA_SCALA_COMPILER_EXPECTED=87
 DOTTY_COMPILER_EXPECTED=81
-SYNTAX_COMPLEXITY_CEILING=2300
+SYNTAX_COMPLEXITY_CEILING=2800
```

Comment:
Do we have an idea of how much is too much here? Like when should it cause refactoring to account for the bump?

Reply:
I guess when we observe compilation time or memory usage to degrade?